Core/Core/Discussions/DiscussionDetailsViewController.swift

```diff
@@ -237,139 +240,104 @@ extension DiscussionDetailsViewController {
+                \(author.map { """
+                    <div class="\(Styles.authorName)" aria-hidden="true">
+                        \(t($0.displayName))
+                    </div>
+                """ } ?? "")
+                \(date.map { """
+                    <div class="\(Styles.date)">\(t($0.dateTimeString))</div>
+                """ } ?? "")
             </div>
+            \(attachment.map { """
+                <a class="\(Styles.blockLink)" href="\(t($0.url?.absoluteString))" aria-label="\(t($0.displayName))">
+                    \(paperclipIcon)
+                </a>
+            """ } ?? "")
         </div>
         """
     }
 
-    static func attachmentLink(_ attachment: File?) -> String {
-        guard let attachment = attachment else { return "" }
+    static func avatarLink(for author: DiscussionParticipant?, isTopic: Bool) -> String {
+        guard let author = author else { return "" }
+        var classes = "\(Styles.avatar)"
+        if isTopic { classes += " \(Styles.avatarTopic)" }
+        var style = ""
+        var content = ""
+        if let url = AvatarView.scrubbedURL(author.avatarURL)?.absoluteString {
+            style += "style=\"background-image:url(\(t(url)))\""
+        } else {
+            content = t(AvatarView.initials(for: author.name))
+            classes += " \(Styles.avatarInitials)"
+        }
         return """
-        <a style="display:flex; align-items:center; font-size:14px; font-weight:600; text-decoration:none; margin:16px 0;" href="\(
-            t(attachment.url?.absoluteString)
-        )">\(
-            paperclipIcon
-        )<span style="-webkit-margin-start:4px; flex:1;">\(
-            t(attachment.displayName)
-        )</span></a>
+        <a class="\(Styles.blockLink)" href="../users/\(author.id)" aria-label="\(t(author.displayName))">
+            <div aria-hidden="true" class="\(classes)" \(style)>\(content)</div>
+        </a>
         """
     }
 
     var topicReplyButton: String {
         guard topic.first?.lockedForUser == false, permissions.first?.postToForum == true else { return "" }
         return """
-        <div style="display:flex; margin:16px 0;">
-            <a href="\(t(topicID))/reply" style="
-                align-items: center;
-                background: var(--brand-buttonPrimaryBackground);
-                border: 0 none;
-                border-radius: 4px;
-                color: var(--brand-buttonPrimaryText);
-                display: flex;
-                padding: 10px 16px;
-                text-decoration: none;
-            ">
-                \(Self.replyIcon)
-                <span style="font-size:14px; font-weight:500; -webkit-margin-start:10px;">
-                    \(t(NSLocalizedString("Reply", bundle: .core, comment: "")))
-                </span>
+        <div style="display:flex; margin:24px 0 16px 0;">
+            <a style="\(Styles.font(.semibold, 16))text-decoration:none" href="\(t(topicID))/reply">
```

Comment:
this element occurs only once, so little is gained by putting the styles in the css sheet. We can still do it for consistency if we want, though.